src/screens/profile-screen/ExpandableBio.tsx

```diff
@@ -0,0 +1,93 @@
+import { useCallback, useState } from 'react'
+
+import { ProfileUser } from 'audius-client/src/pages/profile-page/store/types'
+import {
+  Pressable,
+  View,
+  Text,
+  LayoutChangeEvent,
+  LayoutAnimation
+} from 'react-native'
+import { useToggle } from 'react-use'
+
+import { makeStyles } from 'app/styles/makeStyles'
+
+import { Sites } from './Sites'
+
+const messages = {
+  showMore: 'show more',
+  showLess: 'show less'
+}
+
+const useStyles = makeStyles(({ palette, typography, spacing }) => ({
+  root: {
+    marginTop: spacing(5)
+  },
+  bio: {
+    ...typography.body,
+    color: palette.neutralLight2
+  },
+  expandButton: {
+    marginTop: spacing(2),
+    backgroundColor: 'white'
+  },
+  expandText: {
+    ...typography.h2,
+    color: palette.primary,
+    textTransform: 'capitalize'
+  }
+}))
+
+type ExpandableBioProps = {
+  profile: ProfileUser
+}
+
+export const ExpandableBio = ({ profile }: ExpandableBioProps) => {
+  const { bio, website, donation } = profile
+  const styles = useStyles()
+  const [fullHeight, setFullHeight] = useState(0)
+  const hasSites = Boolean(website || donation) || true
+  const [shouldShowMore, setShouldShowMore] = useState(hasSites)
+  const [isExpanded, setIsExpanded] = useToggle(false)
+
+  const handleBioLayout = useCallback(
+    (event: LayoutChangeEvent) => {
+      const { height } = event.nativeEvent.layout
+      if (!fullHeight) {
+        setFullHeight(height)
+      } else if (fullHeight > height) {
+        setShouldShowMore(true)
+      }
+    },
+    [fullHeight]
+  )
+
+  const handleToggleExpanded = useCallback(() => {
+    LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut)
```

Comment:
Oh this is cool

Reply:
yeah this is the easy way to do animations i learned!